stylegan_code_finder/train_with_lightning.py

```diff
@@ -58,29 +60,33 @@ def main(args: argparse.Namespace):
     logger = WandbLogger(name=args.log_name, save_dir=args.log_dir, project=args.wandb_project_name)
     logging.info("done")
 
-    segmenter_class = get_segmenter_class(config)
-    segmenter = segmenter_class(config)
+    if config['lightning_cpt_pth'] is None:
+        segmenter_class = get_segmenter_class(config)
+        segmenter = segmenter_class(config)
+    else:
+        segmenter = get_segmenter_class(config).load_from_checkpoint(config['lightning_cpt_pth'])
 
     if config['network'] == 'EMANet':
         multiprocessing_strategy = 'ddp'
     else:
         multiprocessing_strategy = 'ddp_find_unused_parameters_false'
 
-    model_checkpoint_callback = ModelCheckpoint(save_top_k=10, monitor='val_loss',
+    model_checkpoint_callback = ModelCheckpoint(save_top_k=3, monitor='val_loss',
                                                 dirpath=Path(config['log_dir'], 'checkpoints'),
                                                 filename="segmentation-{epoch:02d}-{val_loss:.2f}")
     callbacks = [EarlyStopping(monitor='val_loss', patience=config['patience']), model_checkpoint_callback]
 
+    if 'max_iter' in config:
+        config['epochs'] = 1000  # pytorch lightning default
+    else:
+        config['max_iter'] = -1
```

Comment:
Now the code makes sense to me :D  
So your initial code was indeed the better solution and I just don't really understood where the 1000 came from (because it seemed so arbitrary). Your comment cleared it up.  
Sometimes I also make mistakes when reviewing code and I do not fully understand the intentions immediately. In these cases, I'm happy if you explain your thoughts in a comment instead of just blindly enacting my intial feedback :wink: